server/index.js

```diff
@@ -29,4 +35,46 @@ app.get('/shoes/:shoeId/rating', (req, res) => {
   });
 });
 
+//create review
+app.post('/shoes/:shoeId/reviews', (req, res) => {
+  let { shoeId } = req.params;
+  let review = req.body;
+  review.shoe_id = Number(shoeId);
+  controller.addReview(review)
+  .then(data => {
+    res.send(data);
+  })
+  .catch(err => {
+    console.error(err);
+    res.end();
+  });
+});
+
+//delete review
+app.delete('/shoes/reviews/:id', (req, res) => {
+  let { id } = req.params;
+  controller.deleteReview(id)
+  .then(data => {
+    res.send(data);
+  })
+  .catch(err => {
+    console.error(err);
+    res.end();
+  });
+});
+
+//update review
+app.put('/shoes/reviews/:id', (req, res) => {
+  let { id } = req.params;
+  let updatedReview = req.body;
+  controller.updateReview(id, updatedReview)
+  .then(data => {
+    res.send(data);
```

Comment:
What does the data look like that you are sending back? Maybe send a status code.

Reply:
Same thing as delete: { "message": "Updated review with id 1203." }